src/Views/IframeView.php

```diff
@@ -1,88 +1,289 @@
 <?php
 /**
- * Handle form template view skin.
+ * Handle iframe skin.
  *
  * @package Give
  */
 
 namespace Give\Views;
 
+use Give\Form\Template;
+use Give\Helpers\Form\Utils as FormUtils;
+use Give\Helpers\Utils as GlobalUtils;
+
 /**
  * Class IframeView
  *
- * @since 2.7.0
+ * Note: only for internal use.
+ *
+ * @since   2.7.0
  * @package Give
  */
 class IframeView {
 	/**
-	 * Document page title.
+	 * Iframe URL.
 	 *
-	 * This will be use to setup title tag.
+	 * This will be use to setup src attribute.
 	 *
 	 * @since 2.7.0
 	 * @var string
 	 */
-	protected $title;
+	protected $url;
 
 	/**
-	 * Document page body.
-	 *
-	 * This will be use to setup content of body tag.
+	 * Flag to check whether show modal or iframe on page.
 	 *
 	 * @since 2.7.0
-	 * @var string
+	 * @var bool
 	 */
-	protected $body;
+	protected $modal = false;
 
 	/**
-	 * Body classes.
+	 * Flag to check whether on not auto scroll page to iframe.
 	 *
-	 * This will be use to setup body tag classes.
+	 * @since 2.7.0
+	 * @var int
+	 */
+	protected $autoScroll = 0;
+
+	/**
+	 * Iframe minimum height.
 	 *
 	 * @since 2.7.0
-	 * @var array
+	 * @var bool
+	 */
+	protected $minHeight = null;
+
+	/**
+	 * Unique identifier.
+	 *
+	 * @var string|null
+	 */
+	protected $uniqueId = null;
+
+	/**
+	 * Button title.
+	 *
+	 * @var string|null
+	 */
+	protected $buttonTitle = null;
+
+	/**
+	 * Button color.
+	 *
+	 * @var string|null
+	 */
+	protected $buttonColor = null;
+
+	/**
+	 * Form template.
+	 *
+	 * @var Template
+	 */
+	protected $template = null;
+
+	/**
+	 * Form id.
+	 *
+	 * @var int
+	 */
+	protected $formId = 0;
+
+	/**
+	 * IframeView Constructor
+	 *
+	 * @param Template $template
+	 */
+	public function __construct( $template = null ) {
+		$this->uniqueId    = uniqid( 'give-' );
+		$this->buttonTitle = __( 'Click to donate', 'give' );
```

Comment:
Because translated strings are user-defined, the translated string should be escaped.